TreeTracker/UI/Views/AddTree/AddTreeViewController.swift

```diff
@@ -191,3 +233,76 @@ private extension AddTreeViewModel.GPSAccuracy {
+        cameraView = UIView(frame: CGRect(x: 0, y: 0,
+                            width: previewView.bounds.size.width,
+                            height: previewView.bounds.size.height))
+        cameraView.contentMode = .scaleAspectFit
+        cameraView.layer.cornerRadius = 30
+        cameraView.layer.zPosition = 0
+        previewView.addSubview(cameraView)
+    }
+    private func initializeCapture() {
+        session.sessionPreset = .photo
+        guard let device = AVCaptureDevice.default(.builtInWideAngleCamera, for: .video, position: .back) else {
+            return
+        }
+        captureDevice = device
+        beginCapture()
+    }
+    private func beginCapture() {
+        var deviceInput: AVCaptureDeviceInput!
+
+        do {
+            deviceInput = try AVCaptureDeviceInput(device: captureDevice)
+            guard deviceInput != nil else {
+                print("error: cant get deviceInput")
+                return
+            }
+            if self.session.canAddInput(deviceInput) {
+                self.session.addInput(deviceInput)
+            }
+            guard session.canAddOutput(photoOutput) else {return}
+            session.addOutput(photoOutput)
+            session.commitConfiguration()
+            videoDataOutput = AVCaptureVideoDataOutput()
+            videoDataOutput.alwaysDiscardsLateVideoFrames=true
+            videoDataOutputQueue = DispatchQueue(label: "VideoDataOutputQueue")
+            videoDataOutput.setSampleBufferDelegate(self, queue: self.videoDataOutputQueue)
+
+            if session.canAddOutput( self.videoDataOutput ) {
+                session.addOutput( self.videoDataOutput )
+            }
+
+            videoDataOutput.connection(with: .video)?.isEnabled = true
+
+            previewLayer = AVCaptureVideoPreviewLayer(session: self.session)
+            previewLayer.videoGravity = AVLayerVideoGravity.resizeAspectFill
+
+            let rootLayer: CALayer = self.cameraView.layer
+            rootLayer.masksToBounds=true
+            previewLayer.frame = rootLayer.bounds
+            rootLayer.addSublayer(self.previewLayer)
+            session.startRunning()
+        } catch let error as NSError {
+            deviceInput = nil
+            print("error: \(error.localizedDescription)")
+        }
+    }
+
+}
+// MARK: - AddTreeModel AVCapturePhotoDelegate Extension
+extension AddTreeViewController: AVCapturePhotoCaptureDelegate {
+    func didTakePhoto() {
+        let settings = AVCapturePhotoSettings(format: [AVVideoCodecKey: AVVideoCodecType.jpeg])
+        photoOutput.capturePhoto(with: settings, delegate: self)
+    }
+    func photoOutput(_ output: AVCapturePhotoOutput, didFinishProcessingPhoto photo: AVCapturePhoto, error: Error?) {
+        guard let imageData = photo.fileDataRepresentation() else {return}
+        if let editedImage = UIImage(data: imageData) {
+            self.viewModel?.updateTreeImage(treeImage: editedImage)
+        }
```

Comment:
[suggestion:](https://conventionalcomments.org/#labels) Could move both these statements into the guard rather than two statements.
```suggestion
        guard let imageData = photo.fileDataRepresentation(),
            let editedImage = UIImage(data: imageData) else {
                return   
        }
        self.viewModel?.updateTreeImage(treeImage: editedImage)
```